config.dsc

```diff
@@ -41,6 +41,12 @@ config({
             configuration: {
                 toolUrl: "https://dist.nuget.org/win-x86-commandline/v4.9.4/NuGet.exe",
                 hash: "17E8C8C0CDCCA3A6D1EE49836847148C4623ACEA5E6E36E10B691DA7FDC4C39200",
+                credentialProviders: [ 
+					{  
+					  toolUrl: "Shared/Tools/CredentialProviderBuildXL.exe", 

```

Comment:
where is the other credential provider (`Shared/Tools/CredentialProviderDevLocal.VSS.exe`) used?

Reply:
For developer's local builds (Windows log-in)

Reply:
right, but how/where is that specified?

Reply:
It's picked up automatically by nuget since its in the same folder and it's name starts with CredentialProvider*.
It's an existing file, I just renamed it since I want nuget to pick up the BuildXL cred exe first (It picks any CredentialProvider* files besides nuget.exe alphabetically as mentioned here:
https://docs.microsoft.com/en-us/nuget/reference/extensibility/nuget-exe-credential-providers )

Reply:
Had to manually add the new exe since we need it to be copied over to the frontend/Nuget_1/nuget folder for internal (cb and 1es) nuget downloads while building BuildXL

Reply:
Instead of adding the exe, create a nuget package and consume it. Also, check in the CredentialProviderBuildXL source code

---
In reply to: [337805887](https://github.com/microsoft/BuildXL/pull/1087#discussion_r337805887) [](ancestors = 337805887)